packages/frontend/src/components/accounts/two_factor/TwoFactorVerifyModal.js

```diff
@@ -95,15 +97,20 @@ const TwoFactorVerifyModal = ({ open, onClose }) => {
     const handleResendCode = async () => {
         setResendCode('resending');
         await Mixpanel.withTracking('2FA Modal Resend code', 
-            async () => await dispatch(resendTwoFactor()),
+            async () => {
+                await dispatch(resendTwoFactor());
+                setResendCode('resent');
+                setTimeout(() => { setResendCode(); }, 3000);
+            },
             (e) => {
-                setResendCode();
+                if (e.status === TOO_MANY_REQUESTS_STATUS) {
+                    onClose(false, e);
+                } else {
+                    setResendCode();
+                }
+
                 throw e;
             },
-            () => {
-                setResendCode('resent');
-                setTimeout(() => { setResendCode(); }, 3000);
-            }
```

Comment:
We only want to run this on success, but it also gets run on failure: https://github.com/near/near-wallet/blob/master/packages/frontend/src/mixpanel/index.js#L23-L27

I've moved this block to main function so it's only run after a successful resend.